gateway-protocol/src/main/proto/gateway.proto

```diff
@@ -770,4 +780,25 @@ service Gateway {
   rpc ModifyProcessInstance (ModifyProcessInstanceRequest) returns (ModifyProcessInstanceResponse) {
 
   }
+
+  /*
+    Deletes a resource from the state. Once a resource has been deleted it cannot
+    be recovered. If the resource needs to be available again, a new deployment
+    of the resource is required.
+
+    Deleting a process will cancel any running instances of this process
+    definition. Deleting the latest version of a process will cause the previous
+    version to become the latest version.
```

Comment:
🔧 I thought it might be better to explain what it means to a user when a process version is considered 'the latest version'. How about something like this?

```suggestion
    definition. New instances of a deleted process are created using
    the lastest version that hasn't been deleted. Creating a new 
    process instance is impossible when all versions have been
    deleted.
```

🔧 We should also consider mentioning what happens to child instances. It's explicit behavior we want and it might be useful to users to document this.